libgnucash/engine/Account.cpp

```diff
@@ -5991,14 +5991,32 @@ gnc_account_delete_all_bayes_maps (Account *acc)
 /* ================================================================ */
 /* QofObject function implementation and registration */
 
+static void
+destroy_all_child_accounts (Account *acc, gpointer data)
+{
+    xaccAccountBeginEdit (acc);
+    xaccAccountDestroy (acc);
+}
+
 static void
 gnc_account_book_end(QofBook* book)
 {
-    Account *root_account = gnc_book_get_root_account(book);
+    Account *root_account = gnc_book_get_root_account (book);
+    GList *accounts;
+
     if (!root_account)
         return;
-    xaccAccountBeginEdit(root_account);
-    xaccAccountDestroy(root_account);
+
+    accounts = gnc_account_get_descendants (root_account);
+
+    if (accounts)
+    {
+        accounts = g_list_reverse (accounts);
+        g_list_foreach (accounts, (GFunc)destroy_all_child_accounts, nullptr);
+        g_list_free (accounts);
+    }
+    xaccAccountBeginEdit (root_account);
+    xaccAccountDestroy (root_account);
```

Comment:
`xaccAccountDestroy` [should do this already](https://github.com/Gnucash/gnucash/blob/6bf5a618debee47d90317ff287767ba6d1fd32fa/libgnucash/engine/Account.cpp#L1455). If it's broken let's fix that instead of duplicating it.

Reply:
I forgot to change this after I made a similar change for the template root account. Changed locally and it does remove all accounts.